lib/http-client.js

```diff
@@ -0,0 +1,50 @@
+const axios = require('axios');
+const { logRequestError, logRequestInfo, logResponseInfo } = require('./logging');
+
+/** Class for a HTTP-client. */
+class HttpClient {
+  constructor({ baseUrl }) {
+    this.axiosInstance = axios.create({
+      baseURL: baseUrl,
+    });
+
+    this.axiosInstance.defaults.headers.common.Accept = 'application/json+fhir';
+
+    // Dynamically-assigned instance methods for HTTP verbs:
+    ['get', 'post', 'put', 'delete'].forEach((httpVerb) => {
```

Comment:
FHIR also uses `PATCH`. http://hl7.org/fhir/STU3/http.html#patch

Reply:
We'll address the other FHIR verbs in 7-implement-all-fhir-api-verbs